src/cmd/support_archive/logger.go

```diff
@@ -0,0 +1,42 @@
+package support_archive
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	ctrlzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	supportArchiveLoggerName = "[support-archive]"
+)
+
+func registerSupportArchiveLogger(ctx *supportArchiveContext) {
+	if ctx.toStdout {
+		// using stderr because we use stdout to deliver the tarball
+		ctx.log = newSupportArchiveLoggerToWriter(supportArchiveLoggerName, os.Stderr)
+	} else {
+		ctx.log = newSupportArchiveLoggerToWriter(supportArchiveLoggerName, os.Stdout)
+	}
+}
+
+func newSupportArchiveLoggerToWriter(name string, out io.Writer) logr.Logger {
```

Comment:
`newSupportArchiveLoggerToWriter ` this name is reeally confusing, it kinda tells me that I should give it a Logger and I will get a Writer, however it will create a new Logger that is using the provided Writer.

IMO better options: 
`newSupportArchiveLoggerWithWriter`
`newSupportArchiveLogger` as the `Writer` is already part of the params so writing it into the name is a bit redundant.

(if you want more wrapping, but it would just overcomplicate things)
`newSupportArchiveLoggerToStdout`
`newSupportArchiveLoggerToStderr`

Reply:
Yeah, true it's a left-over from when `registerSupportArchiveLogger` still was called `newSupportArchiveLogger` and I needed a distinction. Now I would go with `newSupportArchiveLoggerWithWriter` except we don't need the addendum any more, so `newSupportArchiveLogger` it'll be.